lib/testing/src/lib/process-services/pages/form-fields.page.ts

```diff
@@ -16,144 +16,138 @@
+        await BrowserVisibility.waitUntilElementIsNotVisible(this.formContent);
     }
 
-    checkFormIsDisplayed() {
-        BrowserVisibility.waitUntilElementIsVisible(this.formContent);
-        return this;
+    async checkFormIsDisplayed(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.formContent);
     }
 
-    getNoFormMessage() {
+    async getNoFormMessage(): Promise<string> {
         return BrowserActions.getText(this.noFormMessage);
     }
 
-    getCompletedTaskNoFormMessage() {
+    async getCompletedTaskNoFormMessage(): Promise<string> {
         return BrowserActions.getText(this.completedTaskNoFormMessage);
     }
 
-    clickOnAttachFormButton() {
-        BrowserActions.click(this.attachFormButton);
-        return this;
+    async clickOnAttachFormButton(): Promise<void> {
+        await BrowserActions.click(this.attachFormButton);
     }
 
-    selectForm(formName) {
-        BrowserActions.click(this.selectFormDropDownArrow);
-        BrowserVisibility.waitUntilElementIsVisible(this.selectFormContent);
-        this.selectFormFromDropDown(formName);
-        return this;
+    async selectForm(formName): Promise<void> {
+        await BrowserActions.click(this.selectFormDropDownArrow);
+        await BrowserVisibility.waitUntilElementIsVisible(this.selectFormContent);
+        await this.selectFormFromDropDown(formName);
     }
 
-    selectFormFromDropDown(formName) {
+    async selectFormFromDropDown(formName): Promise<void> {
         const formNameElement = element(by.cssContainingText('span', formName));
-        BrowserActions.click(formNameElement);
+        await BrowserActions.click(formNameElement);
     }
 
-    checkWidgetIsReadOnlyMode(fieldId) {
+    async checkWidgetIsReadOnlyMode(fieldId): Promise<void> {
         const widget = element(by.css(`adf-form-field div[id='field-${fieldId}-container']`));
         const widgetReadOnly = widget.element(by.css('div[class*="adf-readonly"]'));
-        BrowserVisibility.waitUntilElementIsVisible(widgetReadOnly);
-        return widgetReadOnly;
+        await BrowserVisibility.waitUntilElementIsVisible(widgetReadOnly);
     }
 
-    completeForm() {
-        return BrowserActions.click(this.completeButton);
+    async completeForm(): Promise<void> {
+        await BrowserActions.click(this.completeButton);
     }
 
-    setValueInInputById(fieldId, value) {
+    async setValueInInputById(fieldId, value): Promise<void> {
         const input: any = element(by.id(fieldId));
-        BrowserVisibility.waitUntilElementIsVisible(input);
-        input.clear().sendKeys(value);
-        return this;
+        await BrowserVisibility.waitUntilElementIsVisible(input);
+        await BrowserActions.clearSendKeys(input, value);
     }
 
-    isCompleteFormButtonDisabled() {
-        BrowserVisibility.waitUntilElementIsVisible(this.completeButton);
+    async isCompleteFormButtonDisabled(): Promise<string> {
```

Comment:
according to the method name, I would expect a return type of Promise<boolean>

Reply:
not sure I got